### PR TITLE
CSP 666 : Check your answers - Current Job Title

### PR DESCRIPTION
**Pull request by ravihejmadi**
Updated View, Controller &  Tests
CurrentJobTitle Controller & Tests

### CODE REVIEW

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ namespace SFA.DAS.ApprenticeAan.Web.UnitTests.Controllers.Onboarding.CheckYourAn
 public class CheckYourAnswersControllerTests
 {
     [MoqAutoData]
-    public void Index_ReturnsViewResult(
+    public void Index_ChangeJobTitle_ReturnsJobTitleViewResult(
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs:20]: This test changes must be reverted and a new test should be added to check job title change link

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs [hunk 2]
```diff
@@ -33,4 +33,20 @@ public class CheckYourAnswersControllerTests
         result.As<ViewResult>().Model.As<CheckYourAnswersViewModel>().JobTitle.Should().Be(jobTitle);
         result.As<ViewResult>().Model.As<CheckYourAnswersViewModel>().JobTitleChangeLink.Should().Be(TestConstants.DefaultUrl);
     }
+
+    [MoqAutoData]
+    public void Get_ViewModel_BackLink_RedirectsRouteToEmployerDetails(
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs:38]: Remove back link test

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerPostTests.cs [hunk 3]
```diff
@@ -19,14 +19,35 @@ namespace SFA.DAS.ApprenticeAan.Web.UnitTests.Controllers.Onboarding.CurrentJobT
 public class CurrentJobTitleControllerPostTests
 {
     [MoqAutoData]
-    public void Post_SetsEnteredJobTitleInOnBoardingSessionModel(
+        public void Post_SetsEnteredJobTitleInOnBoardingSessionModel(
         [Frozen] Mock<ISessionService> sessionServiceMock,
         [Frozen] Mock<IValidator<CurrentJobTitleSubmitModel>> validatorMock,
         [Greedy] CurrentJobTitleController sut,
         CurrentJobTitleSubmitModel submitmodel,
         OnboardingSessionModel sessionModel)
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerPostTests.cs:27]: all this setup looks like it's indented 1 tab too many, not a big deal tho

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs:1]: This file name should be rename to have `GetTests` suffix

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerPostTests.cs [hunk 4]
```diff
@@ -19,14 +19,35 @@ namespace SFA.DAS.ApprenticeAan.Web.UnitTests.Controllers.Onboarding.CurrentJobT
 public class CurrentJobTitleControllerPostTests
 {
     [MoqAutoData]
-    public void Post_SetsEnteredJobTitleInOnBoardingSessionModel(
+        public void Post_SetsEnteredJobTitleInOnBoardingSessionModel(
```
Comment (MarkFCain) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerPostTests.cs:22]: remove 1 indent (ie put it back how it was)

**Review by MarkFCain — DISMISSED** [review]

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs [hunk 5]
```diff
@@ -47,12 +47,28 @@ public class CurrentJobTitleControllerGetTests
     }
 
     [MoqAutoData]
-    public void Get_ViewModel_HasBackLink(
+    public void Get_BackLink_RedirectsRouteToCheckYourAnswers(
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs:50]: The method name and assertion does not match up. Why rename back link test? Is there a back link test?

**Review by DevenKShah — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.ApprenticeAan.Web/Models/Onboarding/CheckYourAnswersViewModel.cs [hunk 6]
```diff
@@ -4,13 +4,14 @@
 
 namespace SFA.DAS.ApprenticeAan.Web.Models.Onboarding;
 
-public class CheckYourAnswersViewModel
+public class CheckYourAnswersViewModel : IBackLink
```
[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web/Models/Onboarding/CheckYourAnswersViewModel.cs:7]: Removed. Agree

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/Onboarding/PreviousEngagementController.cs [hunk 7]
```diff
@@ -63,6 +63,6 @@ public IActionResult Post(PreviousEngagementSubmitModel submitmodel)
 
         _sessionService.Set(sessionModel);
 
-        return View(ViewPath, model);
+        return RedirectToRoute(RouteNames.Onboarding.CheckYourAnswers);
```
[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web/Controllers/Onboarding/PreviousEngagementController.cs:66]: Moved

File: src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml [hunk 8]
```diff
@@ -1,14 +1,37 @@
...
-    ViewData["Title"] = "Before you start";
+    ViewData["Title"] = "Check your answers";
 }
 
-<div class="govuk-grid-row">
-    <div class="govuk-grid-column-two-thirds">
-        <h1 class="govuk-heading-l">Check the information you have provided before submitting your application</h1>
-        <div id="regions-hint" class="govuk-hint">
-            Here are the details you have provided.
+<form method="post" asp-route="@RouteNames.Onboarding.CheckYourAnswers" enctype="multipart/form-data" novalidate>
```
[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml:8]: Removed

File: src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml [hunk 9]
```diff
@@ -1,14 +1,37 @@
...
+                        <a href=@Url.RouteUrl(RouteNames.Onboarding.CurrentJobTitle) class="govuk-link govuk-link--no-visited-state">
+                            Change
+                        </a>
+                    </dd>
+                </div>
+            </dl>
         </div>
     </div>
-</div>
+    <div class="govuk-button-group">
```
[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml:32]: Removed

File: src/SFA.DAS.ApprenticeAan.Web/Controllers/Onboarding/CheckYourAnswersController.cs [hunk 10]
```diff
@@ -28,6 +28,7 @@ public IActionResult Index()
         _sessionService.Set(sessionModel);
 
         CheckYourAnswersViewModel model = new(Url, sessionModel);
+        model.BackLink = Url.RouteUrl(@RouteNames.Onboarding.PreviousEngagement)!;
```
[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web/Controllers/Onboarding/CheckYourAnswersController.cs:31]: Removed

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs [hunk 11]
```diff
@@ -87,8 +87,7 @@ public class PreviousEngagementControllerPostTests
 
         sut.ModelState.IsValid.Should().BeTrue();
 
-        result.As<ViewResult>().Should().NotBeNull();
-        result.As<ViewResult>().ViewName.Should().Be(PreviousEngagementController.ViewPath);
-        result.As<ViewResult>().Model.As<PreviousEngagementViewModel>().BackLink.Should().Be(TestConstants.DefaultUrl);
+        result.As<RedirectToRouteResult>().Should().NotBeNull();
```
[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/PreviousEngagementControllerTests/PreviousEngagementControllerPostTests.cs:90]: Renamed

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerPostTests.cs [hunk 12]
```diff
@@ -40,13 +61,75 @@ public class CurrentJobTitleControllerPostTests
...
+        submitmodel.JobTitle = null;
+
+        sut.Post(submitmodel);
+
+        sut.ModelState.IsValid.Should().BeFalse();
+    }
+
+    [MoqAutoData]
+    public void Post_BackLink_RedirectsRouteToCheckYourAnswers(
+    [Frozen] Mock<ISessionService> sessionServiceMock,
```
[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerPostTests.cs:108]: Fixed

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerPostTests.cs [hunk 13]
```diff
@@ -40,13 +61,75 @@ public class CurrentJobTitleControllerPostTests
...
+        sessionServiceMock.Verify(s => s.Set(It.Is<OnboardingSessionModel>(m => m.GetProfileValue(ProfileDataId.JobTitle) == submitmodel.JobTitle)));
+
+        result.As<RedirectToRouteResult>().RouteName.Should().Be(RouteNames.Onboarding.CheckYourAnswers);
+    }
+
+
     [MoqAutoData]
     public void Post_Errors_WhenEnteredJobTitleIsNull(
+    [Frozen] Mock<ISessionService> sessionServiceMock,
+    [Greedy] CurrentJobTitleController sut,
```
[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerPostTests.cs:90]: Fixed

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs (hunk 1 above)
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs:20]: Done. This was done after discussing with Yuvraj.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs (hunk 2 above)
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs:38]: Removed

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs:1]: Renamed. Any change to the Controller? (see below). Did discuss this with Yuvraj too
image

Comment (DevenKShah) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerTests.cs:1]: yes that should have `Get` as method name and should have [HttpGet] attribute assigned.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerGetTests.cs [hunk 14]
```diff
@@ -14,13 +14,31 @@
...
+
+        var result = sut.Index();
+
+        sessionServiceMock.Verify(s => s.Set(It.Is<OnboardingSessionModel>(m => m.HasSeenPreview)));
+        result.As<ViewResult>().Model.As<CheckYourAnswersViewModel>().JobTitle.Should().Be(jobTitle);
+        result.As<ViewResult>().Model.As<CheckYourAnswersViewModel>().JobTitleChangeLink.Should().Be(TestConstants.DefaultUrl);
+    }
+
+    [MoqAutoData]
+    public void Index_ChangeJobTitle_ReturnsJobTitleViewResult(
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerGetTests.cs:38]: Method name is slightly misleading... JobTitleChangeLink_LinksToJobTitleRoute

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs [hunk 15]
```diff
@@ -47,12 +47,28 @@ public class CurrentJobTitleControllerGetTests
     }
 
     [MoqAutoData]
-    public void Get_ViewModel_HasBackLink(
+    public void Get_ViewModelHasSeenPreview_RedirectsRouteToCheckYourAnswers(
+    [Frozen] Mock<ISessionService> sessionServiceMock,
+    [Greedy] CurrentJobTitleController sut,
+    OnboardingSessionModel sessionModel)
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs:53]: Indentation is off

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs [hunk 16]
```diff
@@ -47,12 +47,28 @@ public class CurrentJobTitleControllerGetTests
...
+    {
+        sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.Onboarding.CheckYourAnswers);
+        sessionModel.ProfileData.Add(new ProfileModel { Id = ProfileDataId.JobTitle, Value = "Some Title" });
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs:56]: The test name indicates HasSeenPreview is true, but it is not being set here.

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs [hunk 17]
```diff
@@ -47,12 +47,28 @@ public class CurrentJobTitleControllerGetTests
     }
 
     [MoqAutoData]
-    public void Get_ViewModel_HasBackLink(
+    public void Get_ViewModelHasSeenPreview_RedirectsRouteToCheckYourAnswers(
+    [Frozen] Mock<ISessionService> sessionServiceMock,
+    [Greedy] CurrentJobTitleController sut,
+    OnboardingSessionModel sessionModel)
+    {
+        sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.Onboarding.CheckYourAnswers);
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs:55]: There is an default second string argument to AddUrlForRoute, which should be passed to specifically check for that value in the verify

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs [hunk 18]
```diff
@@ -47,12 +47,28 @@ public class CurrentJobTitleControllerGetTests
...
+        result.As<ViewResult>().Model.As<CurrentJobTitleViewModel>().BackLink.Should().Be(TestConstants.DefaultUrl);
+    }
+
+    [MoqAutoData]
+    public void Get_ViewModelHasSeenPreview_RedirectsRouteToEmployerDetails(
         [Frozen] Mock<ISessionService> sessionServiceMock,
         [Greedy] CurrentJobTitleController sut,
         OnboardingSessionModel sessionModel)
     {
         sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.Onboarding.EmployerDetails);
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs:70]: There is an default second string argument to AddUrlForRoute, which should be passed to specifically check for that value in the verify

File: src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml [hunk 19]
```diff
@@ -10,5 +11,20 @@
...
+        <dl class="govuk-summary-list">
+            <div class="govuk-summary-list__row">
+                <dt class="govuk-summary-list__key">
+                                Current job title
+                </dt>
+                <dd class="govuk-summary-list__value">
+                                @Model.JobTitle
+                </dd>
+                <dd class="govuk-summary-list__actions">
+                    <a href=@Url.RouteUrl(RouteNames.Onboarding.CurrentJobTitle) class="govuk-link govuk-link--no-visited-state">
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml:23]: The whole of `<a>` anchor should be in one line for readability.

File: src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml [hunk 20]
```diff
@@ -10,5 +11,20 @@
         <div id="regions-hint" class="govuk-hint">
             Here are the details you have provided.
         </div>
+        <dl class="govuk-summary-list">
+            <div class="govuk-summary-list__row">
+                <dt class="govuk-summary-list__key">
+                                Current job title
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml:17]: What is with all this extra indentations? can be in one line instead?

File: src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml [hunk 21]
```diff
@@ -10,5 +11,20 @@
...
+                </dt>
+                <dd class="govuk-summary-list__value">
+                                @Model.JobTitle
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml:20]: What is with all this extra indentations? can be in one line instead?

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerGetTests.cs [hunk 22]
```diff
@@ -14,13 +14,31 @@
...
+    [Frozen] Mock<ISessionService> sessionServiceMock,
+    [Greedy] CheckYourAnswersController sut,
+    OnboardingSessionModel sessionModel)
```
[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerGetTests.cs:41]: Fixed

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerGetTests.cs [hunk 23]
```diff
@@ -14,13 +14,31 @@
 namespace SFA.DAS.ApprenticeAan.Web.UnitTests.Controllers.Onboarding.CheckYourAnswersControllerTests;
 
 [TestFixture]
-public class CheckYourAnswersControllerTests
+public class CheckYourAnswersControllerGetTests
 {
     [MoqAutoData]
     public void Index_ReturnsViewResult(
     [Frozen] Mock<ISessionService> sessionServiceMock,
```
[the remark this answers is not in the document]
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CheckYourAnswersControllerTests/CheckYourAnswersControllerGetTests.cs:21]: Fixed

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs (hunk 16 above)
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs:56]: It defaults to True. But I am now setting it here and in the Post Tests.

File: src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml (hunk 20 above)
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml:17]: Fixed

File: src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml (hunk 21 above)
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml:20]: Fixed

File: src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml (hunk 19 above)
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web/Views/Onboarding/CheckYourAnswers.cshtml:23]: done

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs (hunk 17 above)
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs:55]: Fixed. Thanks for the help

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs (hunk 18 above)
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs:70]: Fixed. Thanks for the help

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs [hunk 24]
```diff
@@ -48,33 +48,36 @@ public class CurrentJobTitleControllerGetTests
...
+        result.As<ViewResult>().Model.As<CurrentJobTitleViewModel>().BackLink.Should().Be(checkYourAnswersUrl);
     }
 
     [MoqAutoData]
     public void Get_ViewModelHasSeenPreview_RedirectsRouteToEmployerDetails(
         [Frozen] Mock<ISessionService> sessionServiceMock,
         [Greedy] CurrentJobTitleController sut,
-        OnboardingSessionModel sessionModel)
+        OnboardingSessionModel sessionModel,
+        string checkYourAnswersUrl)
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs:71]: currentJobTileUrl

**Review by DevenKShah — DISMISSED** [review]

File: src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs (hunk 24 above)
Comment (ravihejmadi) [reply src/SFA.DAS.ApprenticeAan.Web.UnitTests/Controllers/Onboarding/CurrentJobTitleControllerTests/CurrentJobTitleControllerGetTests.cs:71]: ah! yes!

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**